xbmc/interfaces/Builtins.cpp

```diff
@@ -244,6 +245,42 @@ bool CBuiltins::HasCommand(const std::string& execString)
   return false;
 }
 
+bool CBuiltins::IsSystemPowerdownCommand(const std::string& execString)
+{
+  std::string execute;
+  vector<string> params;
+  CUtil::SplitExecFunction(execString, execute, params);
+  StringUtils::ToLower(execute);
+
+  // Check if action is resulting in system powerdown.
+  if (execute == "reboot"    ||
+      execute == "restart"   ||
+      execute == "reset"     ||
+      execute == "powerdown" ||
+      execute == "hibernate" ||
+      execute == "suspend" )
+  {
+    return true;
+  }
+  else if (execute == "shutdown")
+  {
+    switch (CSettings::Get().GetInt("powermanagement.shutdownstate"))
+    {
+      // Note: Cases taken from CApplicationMessenger::ProcessMessage()
+      case POWERSTATE_SHUTDOWN:
+      case POWERSTATE_SUSPEND:
+      case POWERSTATE_HIBERNATE:
+        return true;
+
+      case POWERSTATE_QUIT:
+      case POWERSTATE_MINIMIZE:
+      case TMSG_RENDERER_FLUSH:
+        return false;
```

Comment:
IMO it makes no sense to duplicate code if we don't need it for this kind of validation. At the latest when a new value is added which is not part of the shutdown commands you need to adjust it on two places, but it is not really necessary.